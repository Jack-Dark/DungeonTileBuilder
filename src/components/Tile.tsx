import React from "react";
import GridSquare from "../components/GridSquare";
import GridHex from "../components/GridHex";

export interface TileProps {
  grid: string;
  content: string;
}

export default class Tile extends React.PureComponent<TileProps> {
  render(): React.ReactNode {
    return (
      <div>
        {/square/i.test(this.props.grid) ? <GridSquare /> : <GridHex />}
      </div>
    );
  }
}
