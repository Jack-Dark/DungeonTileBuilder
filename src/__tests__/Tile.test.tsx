import React from "react";
import Tile from "../components/Tile";
import { shallow, mount, render, ShallowWrapper } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let TileWrapper: ShallowWrapper;
describe("The Tile component", () => {
  beforeEach(() => {
    const props = {};
    TileWrapper = shallow(<Tile {...props} />);
  });
  it("should match the snapshot", () => {
    expect(TileWrapper).toMatchSnapshot();
  });
  describe('when the grid prop is set to "square"', () => {
    beforeEach(() => {
      TileWrapper.setProps({ grid: "square" });
    });
    it("should have a 'GridSquare' component", () => {
      expect(TileWrapper.exists("GridSquare")).toBeTruthy();
    });
  });
  describe('when the grid prop is set to "hex"', () => {
    beforeEach(() => {
      TileWrapper.setProps({ grid: "hex" });
    });
    it("should have a 'GridHex' component", () => {
      expect(TileWrapper.exists("GridHex")).toBeTruthy();
    });
  });
});
