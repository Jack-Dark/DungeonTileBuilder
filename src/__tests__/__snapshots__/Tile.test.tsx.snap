// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Tile component should match the snapshot 1`] = `
<div>
  <GridHex />
</div>
`;
